sdk/src/azure/iot/az_iot_pnp_client_twin.c

```diff
@@ -149,8 +138,7 @@ AZ_NODISCARD az_result az_iot_pnp_client_twin_begin_property_with_status(
 
 AZ_NODISCARD az_result az_iot_pnp_client_twin_end_property_with_status(
     az_iot_pnp_client const* client,
-    az_json_writer* ref_json_writer,
-    az_span component_name)
+    az_json_writer* ref_json_writer)
 {
   _az_PRECONDITION_NOT_NULL(client);
   _az_PRECONDITION_NOT_NULL(ref_json_writer);
```

Comment:
>ref_json_writer [](start = 28, length = 15)

same... we need to do validation of state of json

Reply:
everything in the JSON writer is internal. We will have to let the json API's return an error if there is one

Reply:
I guess depth is something we can depend upon

---
In reply to: [495203192](https://github.com/Azure/azure-sdk-for-c/pull/1363#discussion_r495203192) [](ancestors = 495203192)

Reply:
But similar to my previous answer, is that a limit we want to make? That limits this from being used in some other capacity that the user could want. Ultimately, the json API's themselves will return errors if the JSON is malformed.